Add doc on new field `request.kubernetes_resources` (#48480)
gravitational/teleport: docs/pages/admin-guides/access-controls/access-requests/resource-requests.mdx

in that namespace but you won't be able to access any other supported resources
in the cluster.

> ##### Enforce users to request only allowed Kubernetes Resources

Reviewer: ```suggestion
##### Restrict Access Requests to specific Kubernetes resources
```

> ##### Enforce users to request only allowed Kubernetes Resources

Author: it starts from here https://docs-gtbutz7i0-goteleport.vercel.app/docs/admin-guides/access-controls/access-requests/resource-requests/#restrict-the-resources-a-user-can-request-access-to

and content organization is like:

```
> Restrict the resources a user can request access to
  > kube_cluster
    > Kubernetes resources
      > PREVENTING UNINTENDED ACCESS TO KUBERNETES RESOURCES      
      > ENFORCE USERS TO REQUEST ONLY ALLOWED KUBERNETES RESOURCES 
```

The `request.kubernetes_resources` field allows you to restrict what kinds of Kubernetes 

> resources a user can access request to.

Reviewer: ```suggestion
resources a user can request access to.
```

> The following role assigned to a user, restricts access requesting to only Kubernetes namespaces.

Reviewer: ```suggestion
The following role allows users to request access to Kubernetes namespaces.
```

> The following role assigned to a user, restricts access requesting to only Kubernetes namespaces.
> If a user attempts to make a request to Kubernetes resources other than `namespace` then the request
> will be rejected:

Reviewer 2: The grammar feels a bit off, maybe this is better
```suggestion
The following role, when assigned to a user, would allow them to request access to Kubernetes namespaces.
Requests for Kubernetes resources other than `namespace` will not be allowed.
```

> will be rejected:

Author: i would like to keep it as namespace example as we think that'll be the most often used value

the section above this section explains restricting Kubernetes access.

i added this to the bottom of my section:

```mdx
The `request.kubernetes_resources` field only restricts what `kinds` of Kubernetes resource requests are allowed.
To control Kubernetes access to these resources see 
[Preventing unintended access to Kubernetes resources](#thehash) 
section for more details.
```

```yaml
kind: role
metadata:
  name: requester-kube-access
version: v7
spec:
  allow:
    request:
      search_as_roles:
        - "kube-access"
      kubernetes_resources:
        - kind: "namespace"

>   deny: {}

Reviewer 3: ```suggestion
```

```

> The following role assigned to a user, restricts access requesting to only Kubernetes namespaces and/or pods.

Reviewer 2: ```suggestion
The following role, when assigned to a user, allows access requests for only Kubernetes namespaces and pods.
```

> The following role assigned to a user, restricts access requesting to only Kubernetes namespaces and/or pods.

Reviewer: ```suggestion
The following role allows users to request access only to Kubernetes namespaces and/or pods.
```

```yaml
kind: role
metadata:
  name: requester-kube-access
version: v7
spec:
  allow:
    request:
      search_as_roles:
        - "kube-access"
      kubernetes_resources:
        - kind: "namespace"
        - kind: "pod"

>   deny: {}

Reviewer 3: ```suggestion
```

```

> The following role assigned to a user, restricts users from access requesting to a Kubernetes cluster but
> allows requesting to any Kubernetes resources:

Reviewer 2: ```suggestion
The following role, when assigned to a user, allows them to request access to any specific
Kubernetes resource, but does *not* allow requesting access to the entire cluster.
```

Author: i removed the mention of not allowing requesting access to the entire cluster here, and moved it towards the beginning of the section, see comment: https://github.com/gravitational/teleport/pull/48480/files#r1830229498

Author: btw, you and zac both gave review around same time i think, i took some of zac's suggestion over yours (or tried to combine the best parts)

> The following role assigned to a user, restricts users from access requesting to a Kubernetes cluster but
> allows requesting to any Kubernetes resources:

Reviewer: ```suggestion
The following role restricts users from requesting access to a Kubernetes cluster but
allows requesting to any Kubernetes resources:
```

- Remove "assigned to a user." The sentence reads just as well without it.
- "Access requesting" is not a verb. Let's say "requesting access" instead.
- I don't quite understand the example - how does it restrict users from requesting access to kube clusters?

Author: i removed the `restricting users from requesting access to kube cluster` phrase from this example. instead i moved it to the beginning of section like this:

```
The `request.kubernetes_resources` field allows you to restrict what kinds of Kubernetes 
resources a user can request access to. Configuring this field to any value will disallow
requesting access to the entire Kubernetes cluster.

If the `request.kubernetes_resources` field is not configured, then a user can request access 
to any Kubernetes resources, including the entire Kubernetes cluster.
```

does this make more sense?

Reviewer: Yep!

```yaml
kind: role
metadata:
  name: requester-kube-access
version: v7
spec:
  allow:
    request:
      search_as_roles:
        - "kube-access"
      kubernetes_resources:
        - kind: "*"

>   deny: {}

Reviewer 3: ```suggestion
```

```

> By default, when `request.kubernetes_resources` field is not configured, there are no enforcements and

Reviewer: ```suggestion
If the `request.kubernetes_resources` field is not configured, then
```

> a user can make request to any Kubernetes resources including its cluster.

Reviewer 2: ```suggestion
a user can request access to any Kubernetes resources, including the full cluster.
```

See related section about [Kubernetes Resources](../../../../enroll-resources/kubernetes-access/controls.mdx/#kubernetes_resources) 

> to see a list of supported `kind`. Note that only the `kind` field is supported.

Reviewer: ```suggestion
to see a list of supported `kind` values.
```
